examples/cursor_sync_custom_overlay.py

```diff
@@ -0,0 +1,106 @@
+"""
+Cursor sync via custom overlay
+==============================
+
+Create 2 seperate viewers and mirror the position of the cursor between the
+parent and the child via a vispy overlay.
+
+Read up on the view model here: https://napari.org/stable/guides/napari_models.html
+
+.. tags:: visualization-basic, gui
+"""
+
+import warnings
+
+import numpy as np
+from vispy.scene.visuals import Ellipse
+
+import napari
+from napari._vispy.overlays.base import ViewerOverlayMixin, VispyCanvasOverlay
+from napari._vispy.utils.visual import overlay_to_visual
+from napari.components.overlays import CanvasOverlay
+from napari.utils.events import Event
+
+#
+#   Create a cursor overlay
+#
+
+#   NOTE:
+#   Read up on the view model to understand the reasons for this:
+#   https://napari.org/stable/guides/napari_models.html
```

Comment:
Can we use the sphinx-gallery paragraph syntax (example [here](https://github.com/napari/napari/blob/f19569013a0317f184843968eaab09168cc85b6c/examples/surface_multi_texture_.py#L79-L85)) so that these code blocks render as multiple sections (rendered example [here](https://napari.org/stable/gallery/surface_multi_texture.html#create-the-napari-layers))?

Also, same comment as above, links should be myst-markdown tag references so that they work even if we move pages around.

Reply:
Great call! Done!